# Análisis Exploratorio de Datos - Plan de Compras 2025

El objetivo de este notebook es realizar un análisis exploratorio del Plan de Compras 2025, con el propósito de conocer la estructura del conjunto de datos, identificar valores faltantes, registros duplicados, tipos de variables, distribuciones, posibles valores atípicos y relaciones relevantes entre las variables.

El análisis se desarrolla de forma descriptiva y exploratoria. Los valores atípicos o faltantes no se eliminan automáticamente, ya que primero se debe determinar si responden a la lógica propia del proceso de compras.


## 1. Carga de datos y vista inicial

In [1]:
from pathlib import Path
import pandas as pd

RAIZ = Path.cwd().parents[0]

df_plan_de_compras = pd.read_excel(RAIZ / "plan_de_compras_2025.xlsx" , sheet_name="PLAN DE COMPRAS 2025")

df_plan_de_compras.head()

,Unidad de Compra,ID Proyecto,Tipo Proyecto,Estado Proyecto,Código presupuestario,Nombre Proyecto,Descripción Proyecto,Cantidad de Ítems,Nombre ítem,Tipo compra,...,Cargo responsable,Teléfono responsable,Correo responsable,Fecha de Inicio Compra,Fecha Publicación PAC 2025,Cantidad OC Asociadas Ítem 2025,OC Asociada Item 2025,Item De Arraste,Monto De Arrastre,Meses envio OC de Arraste
0,Ministerio de Vivienda y Urbanismo(766),587-259-PC22,Proyecto estratégico,Publicado,2207-2209-33,Servicio arriendo de una plataforma de gestión...,Servicio de arriendo de una plataforma de gest...,2,Servicio arriendo de una plataforma de gestión...,1,...,Coordinadora de Tecnologías Informáticas | Sis...,2 -2901 1198-,mzunigas@minvu.cl,2022-11-01,2022-11-08,0,NaN,1,20000000,2025-01-01
1,SEREMI MINVU V REGION,632-28-PC22,Proyecto operacional,Publicado,2206-2206001,Servicio de Mantención y Reparaciones Menores ...,Convenio de Suministro para el Servicio de Man...,1,Mantenimiento y Reparación de Edificaciones,1,...,Encargado Sección Administración y Finanzas,32-2186802-,agardaix@minvu.cl,2022-02-01,2022-11-08,1,632-7-SE22,1,4500000,2025-01-01
2,SEREMI MINVU V REGION,632-32-PC22,Proyecto operacional,Publicado,2204001,Servicios de Impresión-Seremi Valparaiso,Servicios de Impresión de Materiales SEREMI y ...,1,Materiales de Oficina,1,...,Encargado Sección Administración y Finanzas,32-2186802-,agardaix@minvu.cl,2022-08-01,2022-11-08,0,NaN,1,3825000,2025-01-01
3,SEREMI MINVU V REGION,632-34-PC22,Proyecto operacional,Publicado,2209006,Servicio de Arriendo de Impresoras Multifuncio...,Servicio de Arriendo de Impresoras Multifuncio...,2,Arriendo de Equipos Informáticos,1,...,Encargado Sección Administración y Finanzas,32-2186802-,agardaix@minvu.cl,2022-08-01,2022-11-08,1,632-13-CC23,1,4500000,2025-01-01
4,SEREMI MINVU V REGION,632-41-PC22,Proyecto operacional,Publicado,2205005,Servicio de Telefonía Fija-Seremi Valparaiso,Servicio de Telefonía Fija SEREMI,2,Telefonía Fija,1,...,Encargado Sección Administración y Finanzas,322186802,agardaix@minvu.cl,2022-12-01,2022-11-08,0,NaN,1,26400000,2025-01-01


### Comentario
La inspección inicial permite verificar que la base contiene información de proyectos, ítems de compra, montos, órdenes de compra, responsables y fechas. Se observan variables de naturaleza categórica, numérica y temporal, por lo que el EDA deberá analizar cada grupo de manera diferenciada.

In [2]:
df_plan_de_compras.sample(5)

,Unidad de Compra,ID Proyecto,Tipo Proyecto,Estado Proyecto,Código presupuestario,Nombre Proyecto,Descripción Proyecto,Cantidad de Ítems,Nombre ítem,Tipo compra,...,Cargo responsable,Teléfono responsable,Correo responsable,Fecha de Inicio Compra,Fecha Publicación PAC 2025,Cantidad OC Asociadas Ítem 2025,OC Asociada Item 2025,Item De Arraste,Monto De Arrastre,Meses envio OC de Arraste
341,Ministerio de Vivienda y Urbanismo(766),1583-138-PC25,Proyecto operacional,Actualizado,2206002,MANTENCION DE VEHICULOS - SECCION MOVILIZACION,MANTENCIÓN PREVENITIVA Y REPARATIVA DE VEHÍCULOS,1,MANTENCION DE VEHICULOS - SECCION MOVILIZACION,4,...,Analista Seccin Movilizacin,22-9011135-,lgascon@minvu.cl,2025-07-01,2025-11-27,23,587-75-SE26,0,0,NaT
336,Ministerio de Vivienda y Urbanismo(766),1583-138-PC25,Proyecto operacional,Actualizado,2206002,MANTENCION DE VEHICULOS - SECCION MOVILIZACION,MANTENCIÓN PREVENITIVA Y REPARATIVA DE VEHÍCULOS,1,MANTENCION DE VEHICULOS - SECCION MOVILIZACION,4,...,Analista Seccin Movilizacin,22-9011135-,lgascon@minvu.cl,2025-07-01,2025-11-27,23,587-866-SE25,0,0,NaT
299,Ministerio de Vivienda y Urbanismo(766),1583-94-PC25,Proyecto operacional,Actualizado,2211002,SERVICIO DE CAPACITACION DIPLOMADOS - BECA 14 ...,Se requiere contratar el servicio de capacitac...,1,SERVICIO DE CAPACITACIÓN DIPLOMADOS - BECA 14 ...,6,...,Jefe(A) Unidad de Capacitacion,29-011468-,cojeda@minvu.cl,2025-04-01,2025-11-27,1,587-294-AG25,0,0,NaT
455,Ministerio de Vivienda y Urbanismo(766),1583-200-PC25,Proyecto operacional,Actualizado,2208007,ADQUISICIÓN PASAJES - SECCION GESTION DE COMPRAS,Cubrir la necesidad de traslado de los funcion...,1,ADQUISICIÓN PASAJES - SECCION GESTION DE COMPRAS,6,...,Enc. Seccion Gestión de Compras,02-29011155-,mmalhueg@minvu.cl,2025-01-01,2025-11-27,321,587-153-CM25,0,0,NaT
300,Ministerio de Vivienda y Urbanismo(766),1583-95-PC25,Proyecto operacional,Actualizado,2211002,SERVICIO DE CAPACITACION DIPLOMADOS - BECA 15 ...,Se requiere contratar el servicio de capacitac...,1,SERVICIO DE CAPACITACIÓN DIPLOMADOS - BECA 15 ...,6,...,Jefe(A) Unidad de Capacitacion,29-011468-,cojeda@minvu.cl,2025-04-01,2025-11-27,1,587-293-AG25,0,0,NaT


La muestra aleatoria complementa la visualización de las primeras filas y ayuda a comprobar que la estructura observada no corresponde únicamente a los primeros registros del archivo.

## 2. Estructura del conjunto de datos

In [3]:
df_plan_de_compras.columns.tolist()

['Unidad de Compra',
 'ID Proyecto',
 'Tipo Proyecto',
 'Estado Proyecto',
 'Código presupuestario',
 'Nombre Proyecto',
 'Descripción Proyecto',
 'Cantidad de Ítems',
 'Nombre ítem',
 'Tipo compra',
 'Cantidad Productos',
 'Monto Unitario Ítem',
 'Monto Total Ítem Año 2025',
 'Cantidad OC',
 'Meses envío OC',
 'Nombre responsable',
 'Cargo responsable',
 'Teléfono responsable',
 'Correo responsable',
 'Fecha de Inicio Compra',
 'Fecha Publicación PAC 2025',
 'Cantidad OC Asociadas Ítem 2025',
 'OC Asociada Item 2025',
 'Item De Arraste',
 'Monto De Arrastre',
 'Meses envio OC de Arraste']

In [4]:
df_plan_de_compras.info()

<class 'pandas.DataFrame'>
RangeIndex: 831 entries, 0 to 830
Data columns (total 26 columns):
 #   Column                           Non-Null Count  Dtype         
---  ------                           --------------  -----         
 0   Unidad de Compra                 831 non-null    str           
 1   ID Proyecto                      831 non-null    str           
 2   Tipo Proyecto                    831 non-null    str           
 3   Estado Proyecto                  831 non-null    str           
 4   Código presupuestario            830 non-null    object        
 5   Nombre Proyecto                  831 non-null    str           
 6   Descripción Proyecto             831 non-null    str           
 7   Cantidad de Ítems                831 non-null    int64         
 8   Nombre ítem                      831 non-null    str           
 9   Tipo compra                      831 non-null    int64         
 10  Cantidad Productos               831 non-null    int64         
 11  Mont

### Comentario
El conjunto de datos contiene **831 registros y 26 variables**. La mayor parte de las columnas se encuentra completa. Existen variables de texto, enteros y fechas. Algunas variables como `Código presupuestario`, `Teléfono responsable` y `Meses envío OC` se almacenan como `object`, lo que es razonable si contienen códigos, formatos mixtos o múltiples valores en una misma celda.

Las variables `Fecha de Inicio Compra`, `Fecha Publicación PAC 2025` y `Meses envio OC de Arraste` ya están reconocidas como fechas, lo que permitirá realizar análisis temporales posteriormente.

In [5]:
df_plan_de_compras.dtypes

Unidad de Compra                              str
ID Proyecto                                   str
Tipo Proyecto                                 str
Estado Proyecto                               str
Código presupuestario                      object
Nombre Proyecto                               str
Descripción Proyecto                          str
Cantidad de Ítems                           int64
Nombre ítem                                   str
Tipo compra                                 int64
Cantidad Productos                          int64
Monto Unitario Ítem                         int64
Monto Total Ítem Año 2025                   int64
Cantidad OC                                 int64
Meses envío OC                             object
Nombre responsable                            str
Cargo responsable                             str
Teléfono responsable                       object
Correo responsable                            str
Fecha de Inicio Compra             datetime64[us]


## 3. Calidad de datos

### 3.1 Valores faltantes

In [6]:
nulos = pd.DataFrame({
    "Cantidad Nulos": df_plan_de_compras.isnull().sum(),
    "Porcentaje Nulos": (
        df_plan_de_compras.isnull().mean() * 100
    ).round(2)
})

nulos = nulos.sort_values(
    "Porcentaje Nulos",
    ascending=False
)

nulos

,Cantidad Nulos,Porcentaje Nulos
Meses envio OC de Arraste,648,77.98
OC Asociada Item 2025,123,14.80
Código presupuestario,1,0.12
Tipo Proyecto,0,0.00
ID Proyecto,0,0.00
Nombre Proyecto,0,0.00
Descripción Proyecto,0,0.00
Cantidad de Ítems,0,0.00
Estado Proyecto,0,0.00
Unidad de Compra,0,0.00


### Comentario
La variable `Meses envio OC de Arraste` presenta **648 valores nulos (77,98%)**, mientras que `OC Asociada Item 2025` presenta **123 valores nulos (14,80%)** y `Código presupuestario` solamente **1 valor nulo (0,12%)**.

El alto porcentaje de nulos de `Meses envio OC de Arraste` no debe interpretarse automáticamente como un problema de calidad. Dado que existe la variable `Item De Arraste`, es posible que el dato solo corresponda a registros que efectivamente sean de arrastre. Por eso conviene comprobar si estos nulos son **estructurales** y coherentes con la lógica de negocio.

In [7]:
# Nulos de "Meses envio OC de Arraste" son coherentes con "Item De Arraste"
pd.crosstab(
    df_plan_de_compras["Item De Arraste"],
    df_plan_de_compras["Meses envio OC de Arraste"].isna(),
    rownames=["Item De Arraste"],
    colnames=["Meses de arrastre es nulo"]
)

Meses de arrastre es nulo,False,True
Item De Arraste,,
0,0,648
1,183,0


In [8]:
# Revisa coherencia entre cantidad de OC asociadas y existencia del identificador de OC
tabla_oc_nulos = pd.crosstab(
    df_plan_de_compras["Cantidad OC Asociadas Ítem 2025"] > 0,
    df_plan_de_compras["OC Asociada Item 2025"].isna(),
    rownames=["Tiene OC asociadas según cantidad"],
    colnames=["OC Asociada Item 2025 es nulo"]
)

tabla_oc_nulos

OC Asociada Item 2025 es nulo,False,True
Tiene OC asociadas según cantidad,,
False,0,123
True,708,0


Esta tabla permite detectar casos donde se informa una cantidad positiva de órdenes de compra asociadas, pero el identificador de la orden se encuentra vacío, o viceversa.

### 3.2 Registros duplicados

In [9]:
cantidad_duplicados = df_plan_de_compras.duplicated().sum()

print(f"Registros duplicados: {cantidad_duplicados}")

Registros duplicados: 0


In [10]:
# Posibles duplicados desde la lógica de negocio
duplicados_negocio = (
    df_plan_de_compras
    .groupby(["ID Proyecto", "Nombre ítem"], dropna=False)
    .size()
    .reset_index(name="Cantidad registros")
    .query("`Cantidad registros` > 1")
    .sort_values("Cantidad registros", ascending=False)
)

duplicados_negocio.head(20)

,ID Proyecto,Nombre ítem,Cantidad registros
88,1583-200-PC25,ADQUISICIÓN PASAJES - SECCION GESTION DE COMPRAS,321
29,1583-138-PC25,MANTENCION DE VEHICULOS - SECCION MOVILIZACION,23
298,1641-53-PC24,SERVICIO DE ASEO Y ESTAFETA,22
306,1654-21-PC23,Servicio Fotocopiadora,14
268,1609-14-PC23,Servicios de aseo,8
185,1583-31-PC25,COMPRA DE ALIMENTOS PERECIBLES SEPTIEMBRE - JA...,8
312,1657-45-PC23,Arriendo de impresora multifuncional,6
138,1583-25-PC25,COMPRA DE ALIMENTOS PERECIBLES MARZO - JARDÍN ...,5
225,1583-5-PC25,MATERIAL DE ASEO-1 - ADMINISTRACIÓN DE INSUMOS,4
180,1583-303-PC23,Mantencion Control de Acceso MINVU (36 meses),4


Los casos anteriores no deben eliminarse automáticamente. Primero se debe comprobar si corresponden a registros legítimos, por ejemplo, distintas órdenes, fechas o condiciones asociadas al mismo ítem.


## 4. Cardinalidad y variables constantes

In [11]:
valores_unicos = df_plan_de_compras.nunique().sort_values()

valores_unicos

Cantidad Productos                   1
Estado Proyecto                      2
Tipo Proyecto                        2
Item De Arraste                      2
Cantidad de Ítems                    3
Fecha Publicación PAC 2025           4
Tipo compra                          6
Cantidad OC                          8
Meses envio OC de Arraste           10
Cantidad OC Asociadas Ítem 2025     11
Unidad de Compra                    16
Fecha de Inicio Compra              41
Meses envío OC                      59
Correo responsable                  64
Código presupuestario               73
Monto De Arrastre                   77
Cargo responsable                   81
Nombre responsable                  85
Teléfono responsable                96
Monto Total Ítem Año 2025          181
Monto Unitario Ítem                210
Descripción Proyecto               297
Nombre ítem                        317
ID Proyecto                        326
Nombre Proyecto                    328
OC Asociada Item 2025    

### Comentario
La cardinalidad muestra que `Cantidad Productos` posee un único valor en toda la base, por lo que no aporta variabilidad al análisis descriptivo. También se observa que existen **326 ID de proyecto distintos frente a 831 registros**, lo que confirma que un proyecto puede estar representado por más de una fila o ítem.

Las variables con pocos valores únicos, como `Estado Proyecto`, `Tipo Proyecto`, `Item De Arraste` y `Tipo compra`, deben interpretarse principalmente como variables categóricas.


In [12]:
# Identificar automáticamente variables constantes
columnas_constantes = [
    columna for columna in df_plan_de_compras.columns
    if df_plan_de_compras[columna].nunique(dropna=False) == 1
]

columnas_constantes


['Cantidad Productos']

## 5. Análisis de variables categóricas

### 5.1 Tipo de proyecto

In [13]:
df_plan_de_compras["Tipo Proyecto"].value_counts()

Tipo Proyecto
Proyecto operacional    613
Proyecto estratégico    218
Name: count, dtype: int64

In [31]:
df_plan_de_compras["Tipo Proyecto"].value_counts(normalize=True).mul(100).round(2)

Tipo Proyecto
Proyecto operacional    73.77
Proyecto estratégico    26.23
Name: proportion, dtype: float64

### Comentario
Los proyectos operacionales concentran **613 registros (73,77%)**, mientras que los proyectos estratégicos representan **218 registros (26,23%)**. La composición del plan está, por tanto, mayoritariamente asociada a proyectos operacionales.

### 5.2 Estado del proyecto

In [14]:
df_plan_de_compras["Estado Proyecto"].value_counts()

Estado Proyecto
Actualizado    648
Publicado      183
Name: count, dtype: int64

In [15]:
df_plan_de_compras["Estado Proyecto"].value_counts(normalize=True).mul(100).round(2)

Estado Proyecto
Actualizado    77.98
Publicado      22.02
Name: proportion, dtype: float64

### Comentario
Se observan **648 registros en estado `Actualizado` y 183 en estado `Publicado`**. Esto equivale aproximadamente a **77,98% y 22,02%**, respectivamente. La base presenta una clara concentración en registros actualizados.

### 5.3 Unidad de compra

In [18]:
df_plan_de_compras["Unidad de Compra"].value_counts()

Unidad de Compra
Ministerio de Vivienda y Urbanismo(766)    684
SEREMI MINVU V REGION                       30
SEREMI MINVU III REGION                     25
SEREMI MINVU VI REGION                      20
SEREMI IV REGION                            14
SEREMI MINVU REGION ARICA Y PARINACOTA       9
SEREMI MINVU XI REGION                       8
SEREMI MINVU II REGION                       8
SEREMI MINVU I REGION                        7
SEREMI VIII REGION                           6
Seremi-Minvu XII Región                      6
SEREMI MINVU VII REGION                      4
SERVICIO DE BIENESTAR MINVU SERVIU           4
SEREMI MINVU RM                              3
SEREMI ÑUBLE-REGIÓN XVI                      2
SEREMI IX REGION                             1
Name: count, dtype: int64

### Comentario
La unidad `Ministerio de Vivienda y Urbanismo(766)` concentra **684 de los 831 registros**, aproximadamente **82,3% del total**. Esto evidencia una alta concentración de observaciones en una sola unidad de compra, mientras que las restantes presentan participaciones mucho menores.

### 5.4 Tipo de compra

In [19]:
df_plan_de_compras["Tipo compra"].value_counts().sort_index()

Tipo compra
1    159
2      4
3     20
4     83
5     79
6    486
Name: count, dtype: int64

In [20]:
df_plan_de_compras["Tipo compra"].value_counts(normalize=True).sort_index().mul(100).round(2)

Tipo compra
1    19.13
2     0.48
3     2.41
4     9.99
5     9.51
6    58.48
Name: proportion, dtype: float64

### Comentario
El código de `Tipo compra` más frecuente es **6**, con **486 registros**, aproximadamente **58,5% del total**. Como la variable está codificada numéricamente, es recomendable disponer del diccionario o descripción de cada código antes de realizar una interpretación sustantiva.

## 6. Análisis de variables monetarias

In [21]:
columnas_montos = [
    "Monto Unitario Ítem",
    "Monto Total Ítem Año 2025",
    "Monto De Arrastre"
]

df_plan_de_compras[columnas_montos].describe()

,Monto Unitario Ítem,Monto Total Ítem Año 2025,Monto De Arrastre
count,8.310000e+02,8.310000e+02,8.310000e+02
mean,1.085214e+08,1.014853e+08,6.139761e+06
std,2.036220e+08,2.024663e+08,3.466941e+07
min,9.000000e+04,0.000000e+00,0.000000e+00
25%,4.200000e+06,3.000000e+06,0.000000e+00
50%,4.000000e+07,3.500000e+07,0.000000e+00
75%,2.020000e+08,2.020000e+08,0.000000e+00
max,4.839000e+09,4.839000e+09,6.747510e+08


In [22]:
df_plan_de_compras[columnas_montos].agg([
    "count",
    "sum",
    "mean",
    "median",
    "min",
    "max"
])

,Monto Unitario Ítem,Monto Total Ítem Año 2025,Monto De Arrastre
count,8.310000e+02,8.310000e+02,8.310000e+02
sum,9.018130e+10,8.433428e+10,5.102141e+09
mean,1.085214e+08,1.014853e+08,6.139761e+06
median,4.000000e+07,3.500000e+07,0.000000e+00
min,9.000000e+04,0.000000e+00,0.000000e+00
max,4.839000e+09,4.839000e+09,6.747510e+08


### Comentario
Las variables monetarias presentan diferencias importantes entre media y mediana. Por ejemplo, `Monto Total Ítem Año 2025` tiene una media cercana a **$101,5 millones** y una mediana de **$35 millones**, mientras que alcanza un máximo de **$4.839 millones**.

Esta diferencia indica una distribución fuertemente asimétrica hacia la derecha: la mayoría de los registros se concentra en montos menores y un grupo reducido de compras de gran magnitud eleva considerablemente el promedio. Por este motivo, la mediana es una medida especialmente útil para representar el monto típico.

In [23]:
for columna in columnas_montos:
    cantidad_negativos = (df_plan_de_compras[columna] < 0).sum()
    print(f"{columna}: {cantidad_negativos} valores negativos")

Monto Unitario Ítem: 0 valores negativos
Monto Total Ítem Año 2025: 0 valores negativos
Monto De Arrastre: 0 valores negativos


No se observan montos negativos en las tres variables monetarias analizadas. Esto elimina una inconsistencia básica, aunque todavía conviene estudiar montos iguales a cero y valores extremadamente altos.

In [24]:
# Cantidad de valores iguales a cero en las variables monetarias
for columna in columnas_montos:
    cantidad_ceros = (df_plan_de_compras[columna] == 0).sum()
    porcentaje_ceros = (df_plan_de_compras[columna] == 0).mean() * 100
    print(f"{columna}: {cantidad_ceros} ceros ({porcentaje_ceros:.2f}%)")

Monto Unitario Ítem: 0 ceros (0.00%)
Monto Total Ítem Año 2025: 16 ceros (1.93%)
Monto De Arrastre: 648 ceros (77.98%)


## 7. Análisis de variables de cantidad

In [25]:
columnas_cantidad = [
    "Cantidad de Ítems",
    "Cantidad Productos",
    "Cantidad OC",
    "Cantidad OC Asociadas Ítem 2025"
]

df_plan_de_compras[columnas_cantidad].describe()

,Cantidad de Ítems,Cantidad Productos,Cantidad OC,Cantidad OC Asociadas Ítem 2025
count,831.000000,831.0,831.000000,831.000000
mean,1.199759,1.0,1.344164,125.756919
std,0.414845,0.0,1.263753,155.041024
min,1.000000,1.0,1.000000,0.000000
25%,1.000000,1.0,1.000000,1.000000
50%,1.000000,1.0,1.000000,4.000000
75%,1.000000,1.0,1.000000,321.000000
max,3.000000,1.0,12.000000,321.000000


### Comentario
`Cantidad Productos` es constante e igual a 1 en los 831 registros, por lo que no aporta variabilidad. `Cantidad de Ítems` se concentra principalmente en 1, con un máximo de 3, y `Cantidad OC` también presenta una fuerte concentración en valores bajos.

En cambio, `Cantidad OC Asociadas Ítem 2025` presenta un comportamiento llamativo: su mediana es **4**, pero el tercer cuartil y el máximo son ambos **321**. Esta concentración en el valor 321 merece una revisión específica para determinar si representa una codificación válida, un valor por defecto o una característica real del proceso.

In [26]:
# Revisar la distribución exacta de la cantidad de OC asociadas
frecuencia_oc_asociadas = (
    df_plan_de_compras["Cantidad OC Asociadas Ítem 2025"]
    .value_counts()
    .sort_index()
)

frecuencia_oc_asociadas

Cantidad OC Asociadas Ítem 2025
0      123
1      206
2       52
3       33
4       24
5        5
7       14
8        8
11      22
23      23
321    321
Name: count, dtype: int64